apollo/hasher.py

```diff
@@ -96,26 +97,73 @@ def __call__(self, record):
                       value=bytearray(wmh[hti * self.band_size:(hti + 1) * self.band_size].data))
 
 
+def modify_feature_scales(batches, arguments):
+    log = logging.getLogger("hash")
+    extractors = dict((ex.NAME, (ex.NAMESPACE, getattr(arguments, "%s_scale" % ex.NAME)))
+                      for ex in __extractors__.values() if "%s_scale" % ex.NAME in dir(arguments))
+
+    for ext in list(extractors):
+        if extractors[ext][1] == 1:
+            del extractors[ext]
+
+    if not extractors:
+        return batches
+
+    err = "You should specify location of docfreq file to modify weights of features"
+    assert arguments.docfreq is not None, err
+    assert os.path.isfile(arguments.docfreq), "docfreq should be a file"
+
+    model = OrderedDocumentFrequencies().load(arguments.docfreq)
+    feature_mapping = model.order
+
+    indices = {}
+    for ext in extractors:
+        namespace = extractors[ext][0]
+        ind = [feature_mapping[k] for k in feature_mapping if k.startswith(namespace)]
+        indices[ext] = ind
+
+    for batch in batches:
+        csc_mat = batch.matrix.tocsc()
```

Comment:
no need for `tocsc()` and the loop. Create the weight vector and multiply CSR by vector